### PR TITLE
Group securitay

### PR DESCRIPTION
**Pull request by dvoet**
provides the ability to change group settings for the purpose of changing access permissions.
https://broadinstitute.atlassian.net/browse/GAWB-3710

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [x] Bump the version in `project/Settings.scala` `createVersion()`
- [x] Update `CHANGELOG.md` for those libraries
- [x] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala [hunk 1]
```diff
@@ -13,11 +14,33 @@ trait GoogleDirectoryDAO {
...
   def isGroupMember(groupEmail: WorkbenchEmail, memberEmail: WorkbenchEmail): Future[Boolean]
   def listGroupMembers(groupEmail: WorkbenchEmail): Future[Option[Seq[String]]]
+  
+  def lockedDownGroupSettings = new GroupSettings()
+    .setWhoCanAdd("ALL_OWNERS_CAN_ADD")
+    .setWhoCanJoin("INVITED_CAN_JOIN")
+    .setWhoCanViewMembership("ALL_MANAGERS_CAN_VIEW")
+    .setWhoCanViewGroup("ALL_OWNERS_CAN_VIEW")
+    .setWhoCanInvite("NONE_CAN_INVITE")
+    .setArchiveOnly("true") //.setWhoCanPostMessage("NONE_CAN_POST")
```
Comment (dvoet) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala]: setting it to archive is the right way to set it so no one can post

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala [hunk 2]
```diff
@@ -13,11 +14,33 @@ trait GoogleDirectoryDAO {
...
-  def createGroup(displayName: String, groupEmail: WorkbenchEmail): Future[Unit]
+  def createGroup(displayName: String, groupEmail: WorkbenchEmail, groupSettings: Option[GroupSettings] = None): Future[Unit]
   def deleteGroup(groupEmail: WorkbenchEmail): Future[Unit]
   def addMemberToGroup(groupEmail: WorkbenchEmail, memberEmail: WorkbenchEmail): Future[Unit]
   def removeMemberFromGroup(groupEmail: WorkbenchEmail, memberEmail: WorkbenchEmail): Future[Unit]
   def getGoogleGroup(groupEmail: WorkbenchEmail): Future[Option[Group]]
   def isGroupMember(groupEmail: WorkbenchEmail, memberEmail: WorkbenchEmail): Future[Boolean]
   def listGroupMembers(groupEmail: WorkbenchEmail): Future[Option[Seq[String]]]
+  
+  def lockedDownGroupSettings = new GroupSettings()
```
Comment (rtitle) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala:25]: I'll take your word on this

**Review by rtitle — APPROVED** [review]
Looks good to me. Good explanation on the inner class -- the approach seems reasonable.

I think travis may be having some issues, I noticed errors on Leo builds too.

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala (hunk 2 above)
Comment (dvoet) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleDirectoryDAO.scala:25]: yeah, there was a bit of trial and error to get it right

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**